### DISCUSSION TITLE
New device can’t add

### DISCUSSION THREAD

**Discussion by phy0930** [make-all/tuya-local#475] [Q&A]
I have a energy recover ventilation device. It's control panel used tyua chips(show tuya iot name is "xfj"). Local tuya and tuya support device list doesn't show it.

**Comment by make-all** [comment]
Other integrations use the Tuya iot category to select a template from which to build the device config.  In the case of the built in integration, it is using cloud API, and the category gives it all it needs to get a basic template working, and it doesn't really go beyond those basics.  I understand localtuya is now using such a template system too, but because it is local needs the extra step to assign dp ids to the functions from the template, and maybe customize some enum values too.

In the case of tuya-local, the configs have always been created for individual devices, using basic building blocks corresponding to Home Assistant entities, so we don't use that "xfj" category at all. Instead, we need information about the device which the "New Device" issue template tries to guide you through collecting.

1. The Log message from the Home Assistant log that is output when you try to add the device (assuming you already got the device id, local key and IP address, and those are working)
2. Information from **one specific** iot.tuya.com API Explorer function, which contains dp_id information linking the ids in the log message to the functions listed by name in the API Explorer docs (the details of which function are in the New Device template, and in  [README.md](https://github.com/make-all/tuya-local#device-support).
3. Product id, model etc info, from another API Explorer function call (there are a few functions that return this, details of one is in the New Device template)
4. Any links to manual or webpages that contain descriptions of the device's functions (especially for non-common device types).  A manual for the Tuya app is ideal (though many just show the pairing process, and not much about the actual function, so a manual for the device's own control panel can sometimes be more useful)

**Reply by phy0930** [reply]
I used Localtuya integration and key-in correct IP, ID and LocalKey. But it shows "Connection to device succeeded but no datapoints found, please try again. Create a new issue and include debug logs if problem persists."
It doesn't show any message about localtuya issue from Home Assistant log after I add new device.

Thank you very much.

**Reply by make-all** [reply]
tuya-local is not localtuya.
